eras/shelley/impl/src/Cardano/Ledger/Shelley/Rules/NewEpoch.hs

```diff
@@ -173,9 +170,11 @@ newEpochTransition = do
           tellReward (DeltaRewardEvent (RupdEvent e event))
           updateRewards ans
         SJust (Complete ru') -> updateRewards ru'
+      let adaPots = totalAdaPotsES es'
```

Comment:
@JaredCorduan  - I'm guessing here is where the subtlety should kick in but didn't!  I choose this EpochState kinda arbitrarily.
The ticket says "after the epoch transition, but before the application of the first block of the epoch.", but I can't tell which bit of the code is doing that.

Reply:
This is a great spot! This rule is only called by the `TICK` rule, which happens before the transactions in the block are applied.

I'm pretty sure, though, that we should use `es` instead of `es'`, so that we are emitting the ada pots events _before_ the reward update is applied. Do you agree, @kderme ?

Reply:
Actually, if not mistaken, I think we want es'''. We want the fees pot to include exactly the fees payed in the previous epoch (and not the fees payed in the new epoch block). The reward pots will be incremented by all rewards payed in the new epoch, by instant rewards and by the deposits refunds, but not affected by withdrawals from the new epoch block.

Reply:
Hoepfully what I'm describing exists :)

Reply:
All four of these epoch states, `es`, `es'`, `es''`, and `es'''` are not affected by withdrawals.

* `es` represents the epoch state at the end of the epoch which is ending
* `es'` represents the epoch state after applying the reward update (so it zeros out the fee pot, manipulates the treasury and the reserves, and pays out the rewards)
* `es''` represents the epoch state after applying the MIR certs.
* `es'''` represents the epoch state after reaping the retiring pools and potentially adjusting the protocol parameters (a change in the deposit amounts would affect the reserves, etc).

Does is still sound to you like you want `es'''`? you can think of it as "the state of the world at the beginning of the next epoch, just before applying transactions".

Reply:
> `es'` represents the epoch state after applying the reward update (so it zeros out the fee pot, ..

Hm is this the case? I though it subtracts the fees payed on epoch - 2, so the fees pot still has the fees payed on the just passed epoch.

Also `es` is the initial state before the block application right? We already have this ledger state so an event would be unnecessary.

Yes, I think I'd like `es'''`

Reply:
> Hm is this the case? I though it subtracts the fees payed on epoch - 2, so the fees pot still has the fees payed on the just passed epoch.

yep, you are correct, good catch. it does not "zero them out" as I said, it just removes the amount listed in the reward update.

>Yes, I think I'd like es'''

great, thanks!